subsys/net/ip/net_if.c

```diff
@@ -1852,6 +1848,12 @@ bool net_if_ipv6_addr_rm(struct net_if *iface, const struct in6_addr *addr)
 			k_mutex_unlock(&lock);
 		}
 
+#if defined(CONFIG_NET_IPV6_DAD)
+		k_mutex_lock(&lock, K_FOREVER);
+		sys_slist_find_and_remove(&active_dad_timers, &ipv6->unicast[i].dad_node);
+		k_mutex_unlock(&lock);
```

Comment:
We have a flag that allows to disable ND protocol on an interface, so it should be checked here:
```suggestion
		if (!net_if_flag_is_set(iface, NET_IF_IPV6_NO_ND)) {
			k_mutex_lock(&lock, K_FOREVER);
			sys_slist_find_and_remove(&active_dad_timers, &ipv6->unicast[i].dad_node);
			k_mutex_unlock(&lock);
		}
```
It wasn't needed when the timer was removed from the `net_if_ipv6_dad_failed()`, as the DAD procedure woudn't even be started in the first place if the flag was set. But now it's no longer the case if we stop the timer directly from `net_if_ipv6_addr_rm()`.

Reply:
Thanks, added! For my understanding, this would only cause unnecessary mutex lock/unlock overhead, right? If DAD hadn't been started because ND is disabled, the `sys_slist_find_and_remove()` call would simply not remove anything.

Reply:
Exactly, it's just to avoid unnecessary actions in this case.